snapshot/bundlestore/groupcache.go

```diff
@@ -61,26 +59,25 @@ func MakeGroupcacheStore(underlying Store, cfg *GroupcacheConfig) (Store, http.H
 	return &groupcacheStore{underlying: underlying, cache: cache}, pool, nil
 }
 
-// Loop will fetch peers and create a list of addresses, including self, to update groupcache.
-func loop(fetcher PeerFetcher, pool *groupcache.HTTPPool) {
-	prevPeers := []string{}
-	for {
-		peerNodes, err := fetcher.Fetch()
-		if err != nil {
-			log.Print("Unable to fetch groupcache peers: ", err)
-			continue
-		}
+// Loop will listen for cluster updates and create a list of peer addresses to update groupcache.
+// Cluster is expected to include the current node.
+func loop(c *cluster.Cluster, pool *groupcache.HTTPPool) {
+	toPeers := func(nodes []cluster.Node) []string {
```

Comment:
I'd make this a function at the top level if you aren't using any variables witin...